src/utils/getMethodDocumentation.js

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2015, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow
+ */
+
+import getFlowType from './getFlowType';
+import getMethodJSDoc from './getMethodJSDoc';
+import getPropertyName from './getPropertyName';
+import getTypeAnnotation from './getTypeAnnotation';
+
+type MethodParameter = {
+  name: string;
+  description?: ?string;
+  type?: ?FlowTypeDescriptor;
+};
+type MethodReturn = {
+  description: ?string;
+  type: ?FlowTypeDescriptor;
+};
+
+type MethodDocumentation = {
+  name: string;
+  description: ?string;
+  modifiers: Array<string>;
+  visibility: string;
+  params: Array<MethodParameter>;
+  return: ?MethodReturn;
+};
+
+function getMethodParamsDoc(methodPath, jsDoc) {
+  const params = [];
+  const functionExpression = methodPath.get('value');
+
+  // Extract param flow types.
+  functionExpression.get('params').each(paramPath => {
+    let type = null;
+    const typePath = getTypeAnnotation(paramPath);
+    if (typePath) {
+      type = getFlowType(typePath);
+    }
+
+    const param = {
+      name: paramPath.node.name,
+      description: null,
+      type,
```

Comment:
This would be nice but like you said it could be in another PR to keep this one reasonable in size. I actually haven't tested destructuring in parameters, I'll see what it does at the moment.